### ISSUE TITLE
Web3 websocket subscription not working

### ISSUE THREAD

**Issue by leovigna** [skalenetwork/skale-dev-sdk#4]
Web3 with a websocket connection does not seem to work with a local sdk chain and the Chainlink Integration testnet chain (wss://chainlink-integration-wss.skale.network) I would expect the subscription to return new blocks here.

I tested with this small snippet:
```typescript
import Web3 from 'web3';

async function main() {
    //Web3
    const web3Http = new Web3(process.env.ETH_HTTP_URL);
    const web3WS = new Web3(process.env.ETH_WS_URL);

    //Block latest
    const latestBlockNumber = await web3Http.eth.getBlockNumber();
    console.debug(latestBlockNumber);
    const latestBlock = await web3Http.eth.getBlock(latestBlockNumber);
    console.debug(latestBlock);

    //Block subscription
    web3WS.eth.subscribe('newBlockHeaders').on('data', block => {
        console.debug(block); //Should log new blocks
    });
}

main();
```

**Comment by sergiy-skalelabs** [comment]
Please try different(more classic) Web3 initialization:

```
let url = require( "url" );
const g_w3mod = require( "web3" );

process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0; // allow self-signed wss and https

function safeURL( arg ) {
    try {
        let sc = arg[0];
        if( sc == "\"" || sc == "'" ) {
            let cnt = arg.length;
            if( arg[cnt - 1] == sc ) {
                let ss = arg.substring( 1, cnt - 1 );
                let objURL = safeURL( ss );
                if( objURL != null && objURL != undefined )
                    objURL.strStrippedStringComma = sc;
                return objURL;
            }
            return null;
        }
        let objURL = url.parse( arg );
        if( ! objURL.hostname )
            return null;
        if( objURL.hostname.length == 0 )
            return null;
        objURL.strStrippedStringComma = null;
        return objURL;
    } catch ( e ) {
        return null;
    }
}

function getWeb3FromURL( strURL ) {
    let w3 = null;
    try {
        const u = safeURL( strURL );
        const strProtocol = u.protocol.trim().toLowerCase().replace( ":", "" ).replace( "/", "" );
        if( strProtocol == "ws" || strProtocol == "wss" ) {
            const w3ws = new g_w3mod.providers.WebsocketProvider( strURL );
            w3 = new g_w3mod( w3ws );
        } else {
            const w3http = new g_w3mod.providers.HttpProvider( strURL );
            w3 = new g_w3mod( w3http );
        }
    } catch ( err ) {
        console.log( "CRITICAL ERROR: Failed to create Web3 connection to " + strURL + ": " + err.toString() );
        w3 = null;
    }
    return w3;
}

... ... ...

const web3Http = getWeb3FromURL(process.env.ETH_HTTP_URL);
const web3WS = getWeb3FromURL(process.env.ETH_WS_URL);
```

**Comment by dimalit** [comment]
@leovigna Looks like that filter is really fired only once, I will check deeper and post results here soon.

**Comment by sergiy-skalelabs** [comment]
WS/wss works.
We need to fix events delivery only.
Here is complete test app:
```
let url = require( "url" );
const g_w3mod = require( "web3" );

process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0; // allow self-signed wss and https

function safeURL( arg ) {
    try {
        let sc = arg[0];
        if( sc == "\"" || sc == "'" ) {
            let cnt = arg.length;
            if( arg[cnt - 1] == sc ) {
                let ss = arg.substring( 1, cnt - 1 );
                let objURL = safeURL( ss );
                if( objURL != null && objURL != undefined )
                    objURL.strStrippedStringComma = sc;
                return objURL;
            }
            return null;
        }
        let objURL = url.parse( arg );
        if( ! objURL.hostname )
            return null;
        if( objURL.hostname.length == 0 )
            return null;
        objURL.strStrippedStringComma = null;
        return objURL;
    } catch ( e ) {
        return null;
    }
}

function getWeb3FromURL( strURL ) {
    let w3 = null;
    try {
        const u = safeURL( strURL );
        const strProtocol = u.protocol.trim().toLowerCase().replace( ":", "" ).replace( "/", "" );
        if( strProtocol == "ws" || strProtocol == "wss" ) {
            const w3ws = new g_w3mod.providers.WebsocketProvider( strURL );
            w3 = new g_w3mod( w3ws );
        } else {
            const w3http = new g_w3mod.providers.HttpProvider( strURL );
            w3 = new g_w3mod( w3http );
        }
    } catch ( err ) {
        console.log( "CRITICAL ERROR: Failed to create Web3 connection to " + strURL + ": " + err.toString() );
        w3 = null;
    }
    return w3;
}

async function main() {
    //Web3
    // const web3Http = new Web3(process.env.ETH_HTTP_URL);
    // const web3WS = new Web3(process.env.ETH_WS_URL);
    
    const web3Http = getWeb3FromURL(process.env.ETH_HTTP_URL);
    const web3WS = getWeb3FromURL(process.env.ETH_WS_URL);
    

    // //Block latest
    const latestBlockNumber = await web3Http.eth.getBlockNumber();
    console.debug(latestBlockNumber);
    const latestBlock = await web3Http.eth.getBlock(latestBlockNumber);
    console.debug(latestBlock);

    //Block subscription
    // web3WS.eth.subscribe('newBlockHeaders').on('data', block => {
    //     console.debug(block); //Should log new blocks
    // });

    subscription_newBlockHeaders = web3WS.eth.subscribe( "newBlockHeaders", function( error, result ) {
        if ( error ) {
            console.log( "w3.eth.subscribe() error:", error );
            process.exit( 0 );
        }
        console.log( "Done, newBlockHeaders subscribed with result: ", result );
    } );
    subscription_newBlockHeaders.on( "data", function( blockHeader ) {
        console.log( "===========================> newBlockHeaders:", blockHeader );
    } );
    subscription_newBlockHeaders.on( "changed", ( x ) => {
        console.log( "===========================> newBlockHeaders changed:", x );
    } );

}

main();
```

**Comment by leovigna** [comment]
Yes the connection seems fine but I'm not receiving any events.

**Comment by dimalit** [comment]
Hi @leovigna! Latest master contains fix (updated skaled docker image 3.4.4). Please check out this version!

**Comment by dimalit** [comment]
Latest fixes are in commit https://github.com/skalenetwork/skale-sdk/commit/e1fe114268bc9b8cf8ba8e32e0cb00b8d6103887
